src/main/java/com/example/weatherwaveapi/service/YandexWeatherService.java

```diff
@@ -0,0 +1,82 @@
+package com.example.weatherwaveapi.service;
+
+import com.example.weatherwaveapi.config.GeneralSettings;
+import com.example.weatherwaveapi.model.request.CoordinateWeatherRequest;
+import com.example.weatherwaveapi.model.request.YandexWeatherRequest;
+import com.example.weatherwaveapi.model.response.yandexapi.YandexApiContainer;
+import com.example.weatherwaveapi.model.response.yandexapi.forecast.YandexForecastResponse;
+import com.example.weatherwaveapi.model.response.yandexapi.weather.YandexWeatherApiResponse;
+import com.example.weatherwaveapi.model.response.yandexapi.weather.YandexWeatherResponse;
+import com.example.weatherwaveapi.util.urlbuilder.OpenWeatherUrlBuilder;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+import static org.springframework.http.HttpMethod.GET;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class YandexWeatherService {
+    private static final String EXCEPTION_MESSAGE = "HTTP error occurred while processing request. Exception message: {}";
+    private static final String STRING_MESSAGE_FORMAT = "Invalid coordinates: %s. %s";
+    private static final String WEATHER_ERROR_MESSAGE = "Failed to retrieve weather data";
+    private static final String FORECAST_ERROR_MESSAGE = "Failed to retrieve forecast data";
+
+    private final RestTemplate restTemplate;
+    private final GeneralSettings settings;
+    private final OpenWeatherUrlBuilder urlBuilder;
```

Comment:
Why OpenWeatherUrlBuilder in YandexWeatherService? Lets rename this class if it is same for both services